fuse_core/include/fuse_core/constraint.h

```diff
@@ -262,29 +263,38 @@ class Constraint
   virtual ceres::CostFunction* costFunction() const = 0;
 
   /**
-   * @brief Create a new Ceres loss function and return a raw pointer to it.
-   *
-   * See http://ceres-solver.org/nnls_modeling.html#lossfunction for a detailed description about loss functions.
-   * Basically, a loss function defines the penalty associated with a specific amount error. By default, or if a NULL
-   * pointer is provided as the loss function, the penalty will be quadratic. This is the loss function associated
-   * with standard least-squares optimization. By providing a different loss function, the penalty for the constraint's
-   * error can be altered.
-   *
-   * This is generally done to reduce the effect of outlier measurements that made it into the optimization problem.
-   * It is always better to remove outliers before they make it into the optimization problem, but no method is perfect.
-   * Using robust loss functions can significantly improve the results and stability of the solution in the presence
-   * of outlier measurements.
+   * @brief Read-only access to the loss function.
    *
    * The Ceres interface requires a raw pointer. Ceres will take ownership of the pointer and promises to properly
    * delete the loss function when it is done. Additionally, Fuse promises that the Constraint object will outlive any
    * generated loss functions (i.e. the Ceres objects will be destroyed before the Constraint objects). This guarantee
    * may allow optimizations for the creation of the loss function objects.
    *
-   * @return A base pointer to an instance of a derived ceres::LostFunction.
+   * @return A base shared pointer to an instance of a derived LostFunction.
+   */
+  Loss::SharedPtr loss() const
+  {
+    return loss_;
+  }
+
+  /**
+   * @brief Set the constraint loss function
+   *
+   * @param[in] loss - The loss function
+   */
+  void loss(const Loss::SharedPtr& loss)
+  {
+    loss_ = loss;
+  }
+
+  /**
+   * @brief Read-only access to the Ceres loss function.
+   *
+   * @return A base pointer to an instance of a derived ceres::LossFunction.
    */
-  virtual ceres::LossFunction* lossFunction() const
+  ceres::LossFunction* lossFunction() const
```

Comment:
I didn't change the name of this method so I didn't have to update the places where it's used. For that reason the name I used for the class that wraps it up is `fuse_core::Loss` instead of `fuse_core::LossFunction`, and similarly, the attribute here is `loss_` instead of `loss_function_`, and same for the `loss()` accessor and mutator. What do you think about this names?

Reply:
I initially call it `LossFunction`, but it's not too different from `Loss` and at the end I took the shorter one. I don't have a strong opinion on this, but I prefer `Loss` over `LossFunction`. I'm all ears for other naming suggestions. :ear:

Reply:
The problem with naming it is that it's a really specific concept, so finding something that doesn't involve the word `loss` is going to be hard. This works for me.